src/systems/sound-effects-system.js

```diff
@@ -0,0 +1,245 @@
+    scene.addEventListener("media-loading", this.onMediaLoading);
+  }
+
+  cleanUp(scene) {
+    scene.removeEventListener("media-loaded", this.onMediaLoaded);
+    scene.removeEventListener("media-loading", this.onMediaLoading);
+  }
+
+  shouldTick() {
+    this.soundsAreReady =
+      this.soundsAreReady ||
+      (this.sounds.tick &&
+        this.sounds.teleportArc &&
+        this.sounds.teleportEnd &&
+        this.sounds.snapRotate &&
+        this.sounds.spawnPen &&
+        this.sounds.penStartDraw &&
+        this.sounds.cameraSnapshot &&
+        this.sounds.enterScene &&
+        this.sounds.quack &&
+        this.sounds.specialQuack &&
+        this.sounds.chatMessage &&
+        this.sounds.freeze &&
+        this.sounds.pin &&
+        this.sounds.mediaLoading &&
+        this.sounds.mediaLoaded);
+    return this.soundsAreReady;
+  }
+
+  tickTeleportSounds(teleporter, state) {
+    if (teleporter.isTeleporting && !state.teleporting) {
+      state.teleportArcSource = playSoundLooped(this.sounds.teleportArc, this.ctx);
+    } else if (!teleporter.isTeleporting && state.teleporting) {
+      state.teleportArcSource.stop();
+      state.teleportArcSource = null;
+      playSound(this.sounds.teleportEnd, this.ctx);
+    }
+    state.teleporting = teleporter.isTeleporting;
+  }
+
+  tick() {
+    if (!this.shouldTick()) return;
+
+    const { leftHand, rightHand, rightRemote } = AFRAME.scenes[0].systems.interaction.state;
+    if (
+      leftHand.held !== this.prevInteractionState.leftHand.held ||
+      rightHand.held !== this.prevInteractionState.rightHand.held ||
+      rightRemote.held !== this.prevInteractionState.rightRemote.held ||
+      (isUI(rightRemote.hovered) && rightRemote.hovered !== this.prevInteractionState.rightRemote.hovered)
+    ) {
+      playSound(this.sounds.tick, this.ctx);
+    }
+    copy(leftHand, this.prevInteractionState.leftHand);
+    copy(rightHand, this.prevInteractionState.rightHand);
+    copy(rightRemote, this.prevInteractionState.rightRemote);
+
+    this.teleporters.leftHand =
+      this.teleporters.leftHand || document.querySelector("#player-left-controller").components.teleporter;
+    this.teleporters.rightHand =
+      this.teleporters.rightHand || document.querySelector("#player-right-controller").components.teleporter;
+    this.teleporters.rightRemote =
+      this.teleporters.rightRemote || document.querySelector("#gaze-teleport").components.teleporter;
+    this.tickTeleportSounds(this.teleporters.leftHand, this.teleporterState.leftHand);
+    this.tickTeleportSounds(this.teleporters.rightHand, this.teleporterState.rightHand);
+    this.tickTeleportSounds(this.teleporters.rightRemote, this.teleporterState.rightRemote);
+
+    const userinput = AFRAME.scenes[0].systems.userinput;
+    if (userinput.get(paths.actions.muteMic)) {
+      playSound(this.sounds.tick, this.ctx);
+    }
+
+    for (let i = 0; i < this.pendingEffects.length; i++) {
```

Comment:
Would be nice for SoundSystem to only do this part and have another system (or maybe even the interaction system) just queue up sounds like everything else. The teleport state feels particularly out of place since its sort of trying to piece together what the teleporter component already knows...